src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/_params.py

```diff
@@ -1005,3 +1010,133 @@ def _configure_security_policy_storage_params(arg_ctx):
         c.extra('vnet_name',
                 options_list=['--vnet-name'],
                 help='The virtual network name')
+
+    ###############################################
+    #                sql managed instance         #
+    ###############################################
+    with self.argument_context('sql managed instance') as c:
+        c.argument('managed_instance_name',
+                   help='The managed instance name',
+                   options_list=['--name', '-n'],
+                   # Allow --ids command line argument. id_part=name is 1st name in uri
+                   id_part='name')
+
+    with self.argument_context('sql managed instance create') as c:
+        # Create args that will be used to build up the ManagedInstance object
+        create_args_for_complex_type(
+            c, 'parameters', ManagedInstance, [
+                'administrator_login',
+                'administrator_login_password',
+                'location',
+                'license_type',
+                'subnet_id',
+                'vcores',
+                'storage_size_in_gb',
+                'sku_name',
+            ])
+
+        c.argument('administrator_login',
+                   options_list=['--admin-user', '-u'],
+                   required=True)
+
+        c.argument('administrator_login_password',
+                   options_list=['--admin-password', '-p'],
+                   required=True)
+
+        c.argument('subnet_id',
+                   required=True,
+                   options_list=['--subnet-id', '-subn'])
```

Comment:
We need to remove or change `-subn`. These abbreviations are supposed to be just 1 character :) Also let's make sure this parameter aligns with `az sql server vnet-rule create` command. You can reused the param validator that the vnet team wrote.

```
    --subnet            [Required]: Name or ID of the subnet that allows access to an Azure Sql
                                    Server. If subnet name is provided, --vnet-name must be
                                    provided.
    --vnet-name                   : The virtual network name.
```

Actually I just realized that `--vnet-name` should just be `--vnet`. I'll make that update.